mappings/net/minecraft/item/Item.mapping

```diff
@@ -1,149 +1,182 @@
 CLASS bbw net/minecraft/item/Item
 	CLASS bbw$a Settings
-		FIELD a fullStackSize I
-		FIELD b durability I
+		FIELD a maxCount I
+		FIELD b maxDamage I
 		FIELD c recipeRemainder Lbbw;
-		FIELD d itemGroup Lbas;
+		FIELD d group Lbas;
 		FIELD e rarity Lbco;
-		FIELD f foodSetting Laxz;
-		METHOD a stackSize (I)Lbbw$a;
-			ARG 1 fullStackSize
+		FIELD f foodSettings Laxz;
+		METHOD a maxCount (I)Lbbw$a;
+			ARG 1 maxCount
 		METHOD a food (Laxz;)Lbbw$a;
-		METHOD a itemGroup (Lbas;)Lbbw$a;
-			ARG 1 itemGroup
+			ARG 1 settings
+		METHOD a group (Lbas;)Lbbw$a;
+			ARG 1 group
 		METHOD a recipeRemainder (Lbbw;)Lbbw$a;
 			ARG 1 recipeRemainder
 		METHOD a rarity (Lbco;)Lbbw$a;
 			ARG 1 rarity
-		METHOD b durabilityIfNotSet (I)Lbbw$a;
-			ARG 1 durability
-		METHOD c durability (I)Lbbw$a;
-			ARG 1 durability
-	FIELD a GETTER_DAMAGED Lbca;
-	FIELD b GETTER_DAMAGE Lbca;
-	FIELD c GETTER_HAND Lbca;
-	FIELD d GETTER_COOLDOWN Lbca;
-	FIELD e GETTER_CUSTOM_MODEL_DATA Lbca;
-	FIELD f BLOCK_ITEM_MAP Ljava/util/Map;
-	FIELD g MODIFIER_DAMAGE Ljava/util/UUID;
-	FIELD h MODIFIER_SWING_SPEED Ljava/util/UUID;
-	FIELD i random Ljava/util/Random;
-	FIELD j itemGroup Lbas;
-	FIELD k PROPERTIES Ljava/util/Map;
+		METHOD b maxDamageIfAbsent (I)Lbbw$a;
+			ARG 1 maxDamage
+		METHOD c maxDamage (I)Lbbw$a;
+			ARG 1 maxDamage
+	FIELD a DAMAGED_PROPERTY_GETTER Lbca;
+	FIELD b DAMAGE_PROPERTY_GETTER Lbca;
+	FIELD c LEFTHANDED_PROPERTY_GETTER Lbca;
+	FIELD d COOLDOWN_PROPERTY_GETTER Lbca;
+	FIELD e CUSTOM_DATA_PROPERTY_GETTER Lbca;
+	FIELD f BLOCK_ITEMS Ljava/util/Map;
+	FIELD g ATTACK_DAMAGE_MODIFIER_UUID Ljava/util/UUID;
+	FIELD h ATTACK_SPEED_MODIFIER_UUID Ljava/util/UUID;
+	FIELD i RANDOM Ljava/util/Random;
+	FIELD j group Lbas;
+	FIELD k propertyGetters Ljava/util/Map;
 	FIELD l rarity Lbco;
-	FIELD m fullStackSize I
-	FIELD n durability I
+	FIELD m maxCount I
+	FIELD n maxDamage I
 	FIELD o recipeRemainder Lbbw;
 	FIELD p translationKey Ljava/lang/String;
-	FIELD q foodSetting Laxz;
+	FIELD q foodSettings Laxz;
 	METHOD <init> (Lbbw$a;)V
 		ARG 1 settings
-	METHOD P_ isMap ()Z
+	METHOD P_ isComplex ()Z
 	METHOD a getTranslationKey ()Ljava/lang/String;
-	METHOD a getAttributeModifiers (Lain;)Lcom/google/common/collect/Multimap;
-		ARG 1 equiptmentSlot
-	METHOD a isInItemGroup (Lbas;)Z
-	METHOD a appendItemsForGroup (Lbas;Lfj;)V
-	METHOD a getRawIdByItem (Lbbw;)I
+	METHOD a getModifiers (Lain;)Lcom/google/common/collect/Multimap;
+		ARG 1 slot
+	METHOD a isIn (Lbas;)Z
+		ARG 1 group
+	METHOD a appendStacks (Lbas;Lfj;)V
+		ARG 1 group
+		ARG 2 stacks
+	METHOD a getRawId (Lbbw;)I
 		ARG 0 item
-	METHOD a onEntityDamaged (Lbcb;Lair;Lair;)Z
+	METHOD a attack (Lbcb;Lair;Lair;)Z
```

Comment:
I don't think you can say a sword attacks a zombie, except as a figure of speech. Would `hurt` make more sense ?

Reply:
Maybe `hit`? It handles stack damaging for the implementations in `MiningToolItem`, `HoeItem`, `SwordItem` and `TridentItem`, then returns true. The default return value in `Item` is `false`.

Reply:
`hit` sounds good to me.

Reply:
Resolved by bf7a33a, after a discussion on Discord